pages/star-search/index.tsx

```diff
@@ -189,7 +190,9 @@ export default function StarSearchPage({ userId, ogImageUrl }: StarSearchPagePro
   const scrollRef = useRef<HTMLDivElement>(null);
   const { feedback, prompt } = useStarSearchFeedback();
   const { toast } = useToast();
+  const { session } = useSession(true);
   const { sessionToken: bearerToken } = useSupabaseAuth();
+  const userId = useMemo(() => (session ? session.id : initialUserId), [session]);
```

Comment:
There will only ever be the currently logged in user here. `useMemo` seems unnecessary as this is not an expensive calculation.